gpytorch/lazy/lazy_tensor.py

```diff
@@ -1794,6 +1853,29 @@ def __getitem__(self, index):
         # We're done!
         return res
 
+    @cached(name="svd")
+    def _svd(self) -> Tuple["LazyTensor", Tensor, "LazyTensor"]:
+        """Method that allows implementing special-cased SVD computation. Should not be called directly"""
+        from gpytorch.lazy.non_lazy_tensor import NonLazyTensor
+
+        U, S, V = torch.svd(self.evaluate())
+        return NonLazyTensor(U), S, NonLazyTensor(V)
+
+    @cached(name="symeig")
+    def _symeig(self, eigenvectors: bool = False) -> Tuple[Tensor, Optional["LazyTensor"]]:
+        """Method that allows implementing special-cased symeig computation. Should not be called directly"""
+        from gpytorch.lazy.non_lazy_tensor import NonLazyTensor
+
+        dtype = self.dtype  # perform decomposition in double precision for numerical stability
+        evals, evecs = torch.symeig(self.evaluate().to(dtype=torch.double), eigenvectors=eigenvectors)
```

Comment:
With this and some of the recent VI PRs, we've started becoming a little prescriptive about what precision ops happen in.  It might be nice for us to find a better solution that hard require people to use fp64 here, since for a lot of cases fp32 will be just fine.

Maybe we could have an internal of registry of methods we run in fp64 that can be user modified? We could accomplish this without directly patching e.g. `torch.symeig` / `torch.cholesky` by defining `gpytorch.symeig` / `gpytorch.cholesky` that get used internally.

Not necessarily important for this PR, but it seems like we could avoid hard coding fp64 casts everywhere.

Reply:
yeah good idea, created an issue for that: #1213

I'll leave it out of this PR for now.